### PR TITLE
Correct Some Broken Links in README from "Mega-Linter" to "MegaLinter"

### LINKED ISSUE #1029: Some README Links Broke During "Mega-Linter" to "MegaLinter" Rename
**Describe the bug**
Until recently "MegaLinter" was known as "Mega-Linter." Most references were renamed, but not all.

**To Reproduce**

1. Navigate to the [README table of contents](https://github.com/megalinter/megalinter#table-of-contents).
2. Click on the links that have MegaLinter in the text (e.g., "Why MegaLinter").
3. Nothing happens.

**Expected behavior**
I expected the page to jump to the corresponding section of the README.

**Additional context**
There may be other cases of this nature (e.g., [mega-linter.report](https://github.com/megalinter/megalinter/blob/f0186c0c02efce6dc19cfbb08550a86d0c82dc9c/.gitignore#L68)). I have not searched exhaustively.

### PR DESCRIPTION
**Pull request by Kurt-von-Laven**
Fixes #1029.

## Proposed Changes

1. Correct some dangling references to "mega-linter." Mega-Linter was previously renamed to MegaLinter, breaking some internal links in the README. Fix said links.
2. Update CSpell links to use main instead of master. The default branch was renamed upstream.
3. Add a missing terminal period in README.

### Author/Contributor
- [x] If documentation is needed for this change, has that been included in this pull request

### Reviewing Maintainer
- [ ] Label as `breaking` if this is a large fundamental change
- [ ] Label as either `automation`, `bug`, `documentation`, `enhancement`, `infrastructure`, or `performance`

### CODE REVIEW

**Comment by nvuillam** [comment]
## [MegaLinter](https://megalinter.github.io) status: :warning: [**WARNING**](https://github.com/megalinter/megalinter/actions/runs/1505803389)

|   Descriptor   |                                                  Linter                                                   |Files|Fixed|                                  Errors                                  |Elapsed time|
|----------------|-----------------------------------------------------------------------------------------------------------|-----|----:|--------------------------------------------------------------------------|------------|
|✅ BASH         |[bash-exec](https://megalinter.github.io/beta/descriptors/bash_bash_exec)                                  |    6|     |                                                                         0|1.26s       |
|✅ BASH         |[shellcheck](https://megalinter.github.io/beta/descriptors/bash_shellcheck)                                |    6|     |                                                                         0|1.49s       |
|✅ BASH         |[shfmt](https://megalinter.github.io/beta/descriptors/bash_shfmt)                                          |    6|    0|                                                                         0|1.28s       |
|✅ COPYPASTE    |[jscpd](https://megalinter.github.io/beta/descriptors/copypaste_jscpd)                                     |yes  |     |no                                                                        |2.5s        |
|✅ CREDENTIALS  |[secretlint](https://megalinter.github.io/beta/descriptors/credentials_secretlint)                         |yes  |     |no                                                                        |3.26s       |
|✅ DOCKERFILE   |[dockerfilelint](https://megalinter.github.io/beta/descriptors/dockerfile_dockerfilelint)                  |    1|     |                                                                         0|0.45s       |
|✅ DOCKERFILE   |[hadolint](https://megalinter.github.io/beta/descriptors/dockerfile_hadolint)                              |    1|     |                                                                         0|0.75s       |
|✅ GIT          |[git_diff](https://megalinter.github.io/beta/descriptors/git_git_diff)                                     |yes  |     |no                                                                        |0.2s        |
|✅ JSON         |[eslint-plugin-jsonc](https://megalinter.github.io/beta/descriptors/json_eslint_plugin_jsonc)              |   20|    0|                                                                         0|2.65s       |
|✅ JSON         |[jsonlint](https://megalinter.github.io/beta/descriptors/json_jsonlint)                                    |   19|     |                                                                         0|9.63s       |
|✅ JSON         |[v8r](https://megalinter.github.io/beta/descriptors/json_v8r)                                              |   20|     |                                                                         0|18.75s      |
|✅ MARKDOWN     |[markdownlint](https://megalinter.github.io/beta/descriptors/markdown_markdownlint)                        |  196|    0|                                                                         0|5.38s       |
|✅ MARKDOWN     |[markdown-link-check](https://megalinter.github.io/beta/descriptors/markdown_markdown_link_check)          |  196|     |                                                                         0|91.56s      |
|✅ MARKDOWN     |[markdown-table-formatter](https://megalinter.github.io/beta/descriptors/markdown_markdown_table_formatter)|  196|    1|                                                                         0|4.73s       |
|✅ OPENAPI      |[spectral](https://megalinter.github.io/beta/descriptors/openapi_spectral)                                 |    1|     |                                                                         0|1.52s       |
|:warning: PYTHON|[bandit](https://megalinter.github.io/beta/descriptors/python_bandit)                                      |  142|     |[**64**](https://github.com/megalinter/megalinter/actions/runs/1505803389)|2.31s       |
|✅ PYTHON       |[black](https://megalinter.github.io/beta/descriptors/python_black)                                        |  142|    0|                                                                         0|2.94s       |
|✅ PYTHON       |[flake8](https://megalinter.github.io/beta/descriptors/python_flake8)                                      |  142|     |                                                                         0|2.51s       |
|✅ PYTHON       |[isort](https://megalinter.github.io/beta/descriptors/python_isort)                                        |  142|    0|                                                                         0|0.51s       |
|✅ PYTHON       |[mypy](https://megalinter.github.io/beta/descriptors/python_mypy)                                          |  142|     |                                                                         0|65.75s      |
|✅ PYTHON       |[pylint](https://megalinter.github.io/beta/descriptors/python_pylint)                                      |  142|     |                                                                         0|18.46s      |
|✅ SPELL        |[cspell](https://megalinter.github.io/beta/descriptors/spell_cspell)                                       |  466|     |                                                                         0|40.95s      |
|✅ SPELL        |[misspell](https://megalinter.github.io/beta/descriptors/spell_misspell)                                   |  312|    0|                                                                         0|0.43s       |
|✅ XML          |[xmllint](https://megalinter.github.io/beta/descriptors/xml_xmllint)                                       |    2|     |                                                                         0|0.21s       |
|✅ YAML         |[prettier](https://megalinter.github.io/beta/descriptors/yaml_prettier)                                    |   77|    0|                                                                         0|4.12s       |
|✅ YAML         |[v8r](https://megalinter.github.io/beta/descriptors/yaml_v8r)                                              |   20|     |                                                                         0|29.95s      |
|✅ YAML         |[yamllint](https://megalinter.github.io/beta/descriptors/yaml_yamllint)                                    |   78|     |                                                                         0|1.49s       |


See errors details in [**artifact MegaLinter reports** on GitHub Action page](https://github.com/megalinter/megalinter/actions/runs/1505803389)

File: docs/all_linters.md [hunk 1]
```diff
@@ -19,7 +19,7 @@
...
-| [**cspell**](https://github.com/streetsidesoftware/cspell){target=_blank}                               |     5.13.1     |     MIT      | [SPELL](descriptors/spell_cspell.md)                                                                                                                                                       |       :heart:       |             [MegaLinter reference](https://github.com/streetsidesoftware/cspell/tree/master/packages/cspell#mega-linter){target=_blank}             |
+| [**cspell**](https://github.com/streetsidesoftware/cspell){target=_blank}                               |     5.13.1     |     MIT      | [SPELL](descriptors/spell_cspell.md)                                                                                                                                                       |       :heart:       |              [MegaLinter reference](https://github.com/streetsidesoftware/cspell/tree/main/packages/cspell#mega-linter){target=_blank}              |
```
Comment (Kurt-von-Laven) [inline docs/all_linters.md:22]: I don't understand why MegaLinter added these two spaces, but I won't question it for now.

Comment (nvuillam) [reply docs/all_linters.md:22]: markdown-table-formatter ;)

**Review by nvuillam — APPROVED** [review]
Great :)

**Pull request merged**